base/string_utils.cpp

```diff
@@ -328,4 +328,25 @@ bool AlmostEqual(string const & str1, string const & str2, size_t mismatchedCoun
   return false;
 }
 
+bool ParseCSVRow(string const & s, vector<string> & target, char const delimiter, size_t const columns)
+{
+  target.clear();
+  using It = TokenizeIterator<SimpleDelimiter, string::const_iterator, true>;
+  for (It it(s, SimpleDelimiter(delimiter)); it; ++it)
+  {
+    string column = *it;
+    strings::Trim(column);
+    target.push_back(column);
+  }
+
+  // Special case: if the string is empty, return an empty array instead of {""}.
+  if (target.size() == 1 && target[0].empty())
+  {
+    target.clear();
+    return false;
+  }
+
+  return columns <= 0 || target.size() == columns;
```

Comment:
Why do you need the first check at all (that columns is less or equal than zero)? What's the problem to parse csv with zero number of columns?